classes/event/grading_response_failed.php

```diff
@@ -0,0 +1,66 @@
+<?php
+// This file is part of the QuestionPy Moodle plugin - https://questionpy.org
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace qtype_questionpy\event;
+
+use moodle_exception;
+use moodle_url;
+
+/**
+ * Grading attempt failed event.
+ *
+ * @package    qtype_questionpy
+ * @author     Jan Britz
+ * @copyright  2024 TU Berlin, innoCampus {@link https://www.questionpy.org}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class grading_response_failed extends \core\event\base {
+    /**
+     * Initialise event parameters.
+     */
+    protected function init() {
+        $this->data['crud'] = 'c';
```

Comment:
Warum `c`? Wird hier etwas erstellt?

Reply:
Ein grading sollte "erstellt" werden. Ich fand `c` passender als die anderen Optionen, wie siehst du das? Ähnliches gilt für unten.

Reply:
Ich denke es geht bei dem Feld darum, was tatsächlich geschehen ist. Wenn das Erstellen fehlgeschlagen ist, dann wurde ja nichts erstellt. Ich denke also `r` wäre am passendsten.

Aber ich sehe auch gerade in der [Doku](https://docs.moodle.org/dev/Events_API#Properties), dass das Feld optional ist. Vielleicht können wir es einfach weglassen.

Reply:
Okay, das Moodle-eigene Event [`email_failed`](https://github.com/moodle/moodle/blob/ec7711b9a60402891db61357f914be161bc3dfbe/lib/classes/event/email_failed.php#L52) hat dort auch `c` stehen, die anderen drei `*_failed`-Events wieder `r`. Ich habe es umgeändert.

Die Doku ist da leider veraltet, es wird eine [`coding_exception`](https://github.com/moodle/moodle/blob/ec7711b9a60402891db61357f914be161bc3dfbe/lib/classes/event/base.php#L719) geschmissen.